### PR TITLE
Add --noout option to prevent the output of ids

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/11515

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-container-restore.1.md [hunk 1]
```diff
@@ -93,6 +93,10 @@ be used once and the restored *container* will have another IP address. This als
 that **--name, -n** cannot be used in combination with **--tcp-established**.\
 *IMPORTANT: This OPTION is only available in combination with **--import, -i**.*
 
+#### **--noout**
+
+Do not print output ids
```
Comment (mheon) [inline docs/source/markdown/podman-container-restore.1.md:98]: Maybe "Do not print any output on success"?

**Comment by Luap99** [comment]
This looks hard to maintain, we always have to remember to honour this option when we are adding new output.

I think it would be simpler if we would make this a global option and use dup2 to set `/dev/null` as stdout.

File: docs/source/markdown/podman-container-restore.1.md (hunk 1 above)
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-container-restore.1.md:98]: I think `Do not print any IDs in the output.`

**Comment by TomSweeneyRedHat** [comment]
I'm a bit confused by this change.  Is it suppressing all IDs throughout?  I.e. image IDs, container id, volume id, manifest id, etc?

**Comment by rhatdan** [comment]
The request was to have no stdout Basically the equivalent of podman * > /dev/null

**Comment by rhatdan** [comment]
@Luap99 interesting idea.

**Comment by giuseppe** [comment]
Could we call it `--quiet`?

At the moment `--quiet` is an option for the pull command, but we are adding it to the main podman scope.

`--noout` as implemented in the PR will also cause `pull --quiet` since we are silencing every write to stdout.

**Comment by rhatdan** [comment]
--quiet in the case of pull will quiet the output of progress, which actually happens on stderr.  This --noout option would stop the output of the container id, which podman pull --quiet does not.
podman --noout pull ...

Will show the progress but no id.

podman --noout pull --quiet ...

Will stop progress and the output of ID.

**Comment by giuseppe** [comment]
> podman --noout pull --quiet ...

would be better to have `podman --quiet=WHAT pull --quiet`?

For now, it can accept just `stdout` that is also the default.  We can extend it later to accept `stderr` or `both`.

**Comment by mheon** [comment]
I'm a little iffy on that - last time we changed a boolean flag to a not-boolean flag (`--systemd=always` being allowed) we definitely broke some people. We *can* if we really want to (4.0 release and all) but I would kind of prefer to avoid that

**Comment by rhatdan** [comment]
@mheon Sure, although 4.0 is the time to break stuff.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by cdoern** [comment]
LGTM, this will fix the scp -q issue as well.

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**